File: sdkwork-ai-sdk-java/src/main/java/com/sdkwork/ai/api/ContextApi.java
package com.sdkwork.ai.api;

import com.sdkwork.ai.http.HttpClient;
import com.sdkwork.ai.model.*;
import java.util.List;
import java.util.Map;

public class ContextApi {
    private final HttpClient client;
    
    public ContextApi(HttpClient client) {
        this.client = client;
    }

    /** Chat with context */
    public ContextResponse chatWith(String context_id, ChatWithContextRequest body) throws Exception {
        return (ContextResponse) client.post(ApiPaths.aiPath("/context/" + context_id + "/chat/completions"), body);
    }

    /** Create context */
    public ContextResponse createContext(ContextCreateRequest body) throws Exception {
        return (ContextResponse) client.post(ApiPaths.aiPath("/context/create"), body);
    }

    /** Get context */
    public ContextResponse getContext(String context_id) throws Exception {
        return (ContextResponse) client.get(ApiPaths.aiPath("/context/" + context_id + ""));
    }

    /** Delete context */
    public Void deleteContext(String context_id) throws Exception {
        client.delete(ApiPaths.aiPath("/context/" + context_id + ""));
        return null;
    }
}

File: sdkwork-ai-sdk-kotlin/src/main/kotlin/com/sdkwork/ai/api/ContextApi.kt
package com.sdkwork.ai.api

import com.sdkwork.ai.*
import com.sdkwork.ai.http.HttpClient

class ContextApi(private val client: HttpClient) {

    /** Chat with context */
    suspend fun chatWith(context_id: String, body: ChatWithContextRequest): ContextResponse? {
        return client.post(ApiPaths.aiPath("/context/$context_id/chat/completions"), body) as? ContextResponse
    }

    /** Create context */
    suspend fun createContext(body: ContextCreateRequest): ContextResponse? {
        return client.post(ApiPaths.aiPath("/context/create"), body) as? ContextResponse
    }

    /** Get context */
    suspend fun getContext(context_id: String): ContextResponse? {
        return client.get(ApiPaths.aiPath("/context/$context_id")) as? ContextResponse
    }

    /** Delete context */
    suspend fun deleteContext(context_id: String): Unit {
        client.delete(ApiPaths.aiPath("/context/$context_id"))
    }
}
